clients/secret/secret_test.go

```diff
@@ -0,0 +1,94 @@
+package secret_test
+
+import (
+	"bytes"
+	"context"
+	"github.com/golang/mock/gomock"
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/clients/secret"
+	"github.com/influxdata/influx-cli/v2/config"
+	"github.com/influxdata/influx-cli/v2/internal/mock"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+	"github.com/stretchr/testify/assert"
+	tmock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSecret_List(t *testing.T) {
+	t.Parallel()
+
+	id, err := influxid.IDFromString("1111111111111111")
+	require.NoError(t, err)
+
+	fakeResults := "data data data"
+
+	testCases := []struct {
+		name string
+		params secret.ListParams
+		defaultOrgName string
+		registerExpectations func(t *testing.T, secretApi *mock.MockSecretsApi)
+	}{
+		{
+			name: "org id",
+			params: secret.ListParams{
+				OrgParams: clients.OrgParams{
+					OrgID: id,
+				},
+			},
+			defaultOrgName: "default-org",
+			registerExpectations: func(t *testing.T, secretApi *mock.MockSecretsApi) {
+				secretApi.EXPECT().GetOrgsIDSecrets(gomock.Any(), gomock.Any()).Return(api.ApiGetOrgsIDSecretsRequest{ApiService: secretApi})
+				secretApi.EXPECT().GetOrgsIDSecretsExecute(tmock.MatchedBy(func(in api.ApiGetOrgsIDSecretsRequest) bool {
+					body := in.GetOrgID()
+					return assert.NotEmpty(t, body) &&
+						assert.Equal(t, body, id.String())
+				})).Return(api.SecretKeysResponse{Secrets: &[]string{fakeResults}}, nil)
```

Comment:
Since we expect `Execute` to take the request generated by `GetOrgsIDSecrets` w/ no changes, this can be simplified:
```suggestion
				req := api.ApiGetOrgsIDSecretsRequest{ApiService: secretApi}.OrgID(id)
				secretApi.EXPECT().GetOrgsIDSecrets(gomock.Any(), gomock.Eq(id)).Return(req)
				secretApi.EXPECT().GetOrgsIDSecretsExecute(gomock.Eq(req)).Return(api.SecretKeysResponse{Secrets: &[]string{fakeResults}}, nil)
```